### ISSUE TITLE
gpmem: storage of observed values with f_compute vs f_emulate

### ISSUE THREAD

**Issue by Schaechtle** [Schaechtle/matrixGP#1]
we should store values observed with f_emulate and predicted with f_compute in separate places.

**Comment by bzinberg** [comment]
What's your reasoning?

**Comment by Schaechtle** [comment]
A friend could tell us about the value of the function, which we can 
model by observe f_emu(bar) = foo.  This could disagree with predict 
f_compute(bar).

Pointed out by VKM yesterday.


On 20/11/15 17:50, bzinberg wrote:
> What's your reasoning?
> On Nov 20, 2015 9:43 AM, "Schaechtle" <notifications@github.com> wrote:
>
> > we should store values observed with f_emulate and predicted with
> > f_compute in separate places.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/Schaechtle/matrixGP/issues/1>.
> >
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/Schaechtle/matrixGP/issues/1#issuecomment-158473170>.
>

-- 
www.schaechtle.com

**Comment by bzinberg** [comment]
True, but then again, two different friends could disagree on the value of
the function too.  Assuming all computations and observations are accurate
and noiseless is the easiest framework, but one could imagine having gpmem
backed by a probabilistic truth maintenance system (look up TMS if not
heard of it) -- an anomaly detection system that stochastically retracts
observations in an intelligent way.  Could you add me to the loop of your
discussions of gpmem with vkm?

**Comment by Schaechtle** [comment]
fair enough. This was discussed during a research meeting where I joined 
via skype. Not sure how I can loop you in those. I can echo your 
argument in Flowdock's main thread (or you can do it yourself, if you wish).

On 20/11/15 19:24, bzinberg wrote:
> True, but then again, two different friends could disagree on the value of
> the function too. Assuming all computations and observations are accurate
> and noiseless is the easiest framework, but one could imagine having gpmem
> backed by a probabilistic truth maintenance system (look up TMS if not
> heard of it) -- an anomaly detection system that stochastically retracts
> observations in an intelligent way. Could you add me to the loop of your
> discussions of gpmem with vkm?
> On Nov 20, 2015 9:57 AM, "Schaechtle" <notifications@github.com> wrote:
>
> > A friend could tell us about the value of the function, which we can
> > model by observe f_emu(bar) = foo. This could disagree with predict
> > f_compute(bar).
> >
> > Pointed out by VKM yesterday.
> >
> >
> > On 20/11/15 17:50, bzinberg wrote:
> > > What's your reasoning?
> > > On Nov 20, 2015 9:43 AM, "Schaechtle" <notifications@github.com> 
> wrote:
> > >
> > > > we should store values observed with f_emulate and predicted with
> > > > f_compute in separate places.
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <https://github.com/Schaechtle/matrixGP/issues/1>.
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > 
> <https://github.com/Schaechtle/matrixGP/issues/1#issuecomment-158473170
> > >.
> > >
> >
> > --
> > www.schaechtle.com
> >
> > —
> > Reply to this email directly or view it on GitHub
> > 
> <https://github.com/Schaechtle/matrixGP/issues/1#issuecomment-158474766>.
> >
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/Schaechtle/matrixGP/issues/1#issuecomment-158500630>.
>

-- 
www.schaechtle.com

**Comment by axch** [comment]
For what it's worth, I disagree with vkm's reasoning on this point.  If you want observation noise, you can always write (observe (normal (f_emu x) foo) ...).  If there is an efficiency problem with this plan, then one can imagine a (collapsed?) gpmem variant whose model reflects some such noise story, but that's not the same object as gpmem, and gpmem should not be twisted into being it.

**Comment by bzinberg** [comment]
Yep, that's what I was saying on Flowdock.
